wordpress-mobile/WordPress-Android
# Mobile Stories block

This PR adds support for the jetpack Story block to be rendered on gutenberg-mobile, and provides the bridge implementation on the WPAndroid side of things to connect the block actions to the Story composer.

Related Jetpack PR: https://github.com/Automattic/jetpack/pull/17140
Related Gutenberg PR: https://github.com/WordPress/gutenberg/pull/25242
Related Gutenberg-mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2611


#### test block rendering
1. create a story
2. publish it
3. now go to the posts list
4. open it
5. observe the block appears (it's not unsupported but an actual story block).

#### Test block editing
CASE A: the story is in memory (no prefs are used)
1. create a story
2. publish it
3. open the post on gutenberg
4. tap on the block's tool icon
5. it should load on the StoryComposer from memory

CASE B: the story has been created on this device, not in memory
1. create a story
2. publish it
3. kill the app and open it again
4. open the post on gutenberg
5. tap on the block
6. it should load on the StoryComposer from reviving it from prefs (added views should be editable)

CASE C: the story has not been created on this device
1. create a story
2. publish it
3. log in on another device (or uninstall the app and then re-install it)
4. open the post on gutenberg
5. tap on the block
6. it should load on the StoryComposer with the flattened views only (no added views, only background media)

CASE C1: story created on another device, media not present on current device
1. go to a device A and create a story, publish it
2. go to a device B and, WITHOUT refreshing the media section of the site, go straight to Posts list -> refresh (or wait) and tap on the newly created Post
3. observe the post loads
4. when you tap on the tool icon for the story block, it gets loaded and the "Limited story editing" dialog appears.

CASE C2: story created on another device, media not present on current device, NO NETWORK
1. go to a device A and create a story, publish it
2. go to a device B and, WITHOUT refreshing the media section of the site, go straight to Posts list -> refresh (or wait) until it appears on the list
3. turn airplane mode ON
4. tap on the newly created Post (if you're debugging, a red screen will appear on Gutenberg because it can't connect to metro - at this point, turn airplane mode OFF again, as it would have gotten past the media fetching code, now tap on the "Reload" option at the bottom right of the screen and allow the block to render the remote site http URL even if MediaModels haven't been fetched)
5. observe the post loads
6. when you tap on the tool icon for the story block, it cannot load and the following Dialog appears.

<img width="394" alt="Screen Shot 2020-09-18 at 09 53 51" src="https://user-images.githubusercontent.com/6597771/93600448-086d1a80-f996-11ea-9aad-9c74a155b6e0.png">




PR submission checklist:
